ci-operator/step-registry/aws/provision/dns-firewall/aws-provision-dns-firewall-commands.sh

```diff
@@ -0,0 +1,186 @@
+  "api.openshift.com",
+
+  "api.pagerduty.com",
+  "events.pagerduty.com",
+  "api.deadmanssnitch.com",
+  "nosnch.in",
+  "osdsecuritylogs.splunkcloud.com",
+  "http-inputs-osdsecuritylogs.splunkcloud.com",
+  "sftp.access.redhat.com"
+')
+
+DOMAIN_LIST=${OCP_DOMAIN_ALLOW_LIST}
+if [[ ${MANAGED_CLUSTER} == "yes" ]]; then
+  DOMAIN_LIST=${OSD_DOMAIN_ALLOW_LIST}
+fi
+
+function aws_add_param_to_json() {
+    local k="$1"
+    local v="$2"
+    local param_json="$3"
+    if [ ! -e "$param_json" ]; then
+        echo -n '[]' > "$param_json"
+    fi
+    cat <<< "$(jq  --arg k "$k" --arg v "$v" '. += [{"ParameterKey":$k, "ParameterValue":$v}]' "$param_json")" > "$param_json"
+}
+
+# Create DNS firewall
+UNIQUE_HASH=$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 4)
+STACK_NAME="${NAMESPACE}-${UNIQUE_HASH}-dns-firewall"
+dns_firewall_domain_list_name="${STACK_NAME}-domain-list"
+dns_firewall_rule_group_name="${STACK_NAME}-rule-group"
+RULE_GROUP_ALLOWED=$(echo -e '
+{
+    "AWSTemplateFormatVersion": "2010-09-09",
+    "Description": "",
+    "Resources": {
+        "ResolverFirewallDomainList": {
+            "Type": "AWS::Route53Resolver::FirewallDomainList",
+            "Properties": {
+                "Name": "'${dns_firewall_domain_list_name}'",
+                "Domains": ['${DOMAIN_LIST}']
+            }
+        },
+        "ResolverFirewallRuleGroup": {
+            "Type": "AWS::Route53Resolver::FirewallRuleGroup",
+            "Properties": {
+                "Name": "'${dns_firewall_rule_group_name}'",
+                "FirewallRules": [
+                    {
+                        "Priority": 1,
+                        "Action": "ALLOW",
+                        "FirewallDomainListId": {
+                            "Ref": "ResolverFirewallDomainList"
+                        }
+                    }
+                ]
+            }
+        }
+    },
+    "Parameters": {},
+    "Metadata": {},
+    "Conditions": {}
+}')
+
+echo "Create DNS firewall"
+aws --region "${REGION}" cloudformation create-stack \
+  --stack-name "${STACK_NAME}" \
+  --template-body "$(echo $RULE_GROUP_ALLOWED | jq -c)" &
+
+wait "$!"
+echo "Created stack"
+
+aws --region "${REGION}" cloudformation wait stack-create-complete --stack-name "${STACK_NAME}" &
+wait "$!"
+echo "Waited for stack"
+echo "${STACK_NAME}" > "${SHARED_DIR}/dns_firewall_stack_name"
+
+# Save stack information to ${SHARED_DIR} for deprovision step
+echo ${STACK_NAME} >> "${SHARED_DIR}/to_be_removed_cf_stack_list"
+
+# Associate the VPC to the DNS firewall
+vpc_id=$(head -n 1 ${SHARED_DIR}/vpc_id)
+if [[ -z $vpc_id ]]; then
+  echo "Error: Can not get VPC id, exit"
+  echo "vpc: $vpc_id"
+  exit 1
+fi
+
+echo "Associate the VPC $vpc_id to the DNS firewall rule group $dns_firewall_rule_group_name"
+RuleGroupID=$(aws --region "${REGION}" route53resolver list-firewall-rule-groups | jq -r --arg k $dns_firewall_rule_group_name '.FirewallRuleGroups[] |select(.Name==$k) .Id')
```

Comment:
rule group id can be easily retrieved from CF output (or aws command if you decide to use aws util instead.)